ganttproject/src/net/sourceforge/ganttproject/chart/ResourceLoadRenderer.java

```diff
@@ -68,200 +49,73 @@ public ResourceLoadRenderer(ChartModelResource model, ResourceChart resourceChar
-    final Text loadLabel = myTextCanvas.createText(rect.getMiddleX(), rect.getTopY(), "");
-    loadLabel.setSelector(new TextSelector() {
       @Override
-      public Label[] getLabels(TextMetrics textLengthCalculator) {
-        int loadInt = Math.round(load.load);
-        String loadStr = loadInt + "%";
-        int emsLength = textLengthCalculator.getTextLength(loadStr);
-        boolean displayLoad = (loadInt != 100 && emsLength <= rect.getWidth());
-        return displayLoad ? new Label[] {loadLabel.createLabel(loadStr, rect.getWidth())} : new Label[0];
+      public int getRowHeight() {
+        return getConfig().getRowHeight();
       }
-    });
-    loadLabel.setAlignment(HAlignment.CENTER, VAlignment.TOP);
-    loadLabel.setStyle("text.resource.load");
-  }
-
-  private Rectangle createRectangle(List<Offset> offsets, Date start, Date end, int ypos) {
-    if (start.after(getChartEndDate()) || end.compareTo(getChartStartDate()) <= 0) {
-      return null;
-    }
-    OffsetLookup offsetLookup = new OffsetLookup();
-    int[] bounds = offsetLookup.getBounds(start, end, offsets);
-    return getPrimitiveContainer().createRectangle(bounds[0], ypos, bounds[1] - bounds[0], getConfig().getRowHeight());
-  }
-
-  private Date getChartStartDate() {
-    return getChartModel().getStartDate();
-  }
-
-  private Date getChartEndDate() {
-    return getChartModel().getBottomUnitOffsets().get(getChartModel().getBottomUnitOffsets().size() - 1).getOffsetEnd();
-  }
 
-  private List<Offset> getDefaultOffsets() {
-    return getChartModel().getDefaultUnitOffsets();
-  }
-
-  public void beforeProcessingTimeFrames() {
-    myDistributions = new ArrayList<LoadDistribution>();
-    getPrimitiveContainer().clear();
-    HumanResource[] resources = ((ChartModelResource) getChartModel()).getVisibleResources();
-    for (HumanResource resource : resources) {
-      LoadDistribution nextDistribution = resource.getLoadDistribution();
-      myDistributions.add(nextDistribution);
-    }
+      @Override
+      public int getYCanvasOffset() {
+        return getConfig().getHeaderHeight() - myModel.getVerticalOffset();
+      }
+    };
   }
 
-  /**
-   * Class to use as Model object to display the load percentage in the
-   * rectangle.
-   *
-   * @author bbaranne
-   */
-  static class ResourceLoad {
-    private float load;
-
-    ResourceLoad(float load) {
-      this.load = load;
-    }
-
-    public float getLoad() {
-      return load;
+  private List<CapacityHeatmapSceneBuilder.Resource> getResources() {
+    HumanResource[] humanResources = ((ChartModelResource) getChartModel()).getVisibleResources();
+    List<CapacityHeatmapSceneBuilder.Resource> resources = new ArrayList<>();
+
+    for (HumanResource humanResource : humanResources) {
+      List<CapacityHeatmapSceneBuilder.Load> loads = new ArrayList<>();
+      List<Load> tasksLoads = humanResource.getLoadDistribution().getTasksLoads();
```

Comment:
Apparently LoadDistribution will do the same stuff which we do in biz.ganttproject.core.chart.scene.CapacityHeatmapSceneBuilder#calcLoadDistribution ? Can you add TODO to remove this work duplication?

Reply:
True, but we can't use LoadDistribution in GP cloud. Should I remove that duplication in LoadDistribution then?

Reply:
Yes, if it is no more useful for GP desktop purposes.